### PR TITLE
open-cluster-management/backlog#11116--Subscribing specific branch an…

### PR DESCRIPTION
**Pull request by swopebe**
…d directory of Git repository

### CODE REVIEW

File: applications/subscribe_git_resources.adoc [hunk 1]
```diff
@@ -240,6 +241,72 @@ data:
...
+[#subscribing-specific-git]
+== Subscribing specific Git elements
+
+You can subscribe to a specific Git branch, commit, or tag.
+
+[#subscribing-specific-branch]
+=== Subscribing to a specific branch
+
+The subscription operator that is included in this `multicloud-operators-subscription` repository subscribes to the `master` branch of a Git repository by default. If you want to subscribe to a different branch, you need to specify the branch name annotation in the subscription.
+
```
Comment (swopebe) [inline applications/subscribe_git_resources.adoc:253]: @rokej any idea where we are with changing master branch to `main`?

Comment (swopebe) [reply applications/subscribe_git_resources.adoc:253]: @dockerymick can you just peer review what has changed?

Comment (dockerymick) [reply applications/subscribe_git_resources.adoc:253]: I suggest changing "this" to "the" and agree to update master to main

File: applications/subscribe_git_resources.adoc [hunk 2]
```diff
@@ -240,6 +241,72 @@ data:
...
+The following example, the YAML file displays how to specify a different branch with `apps.open-cluster-management.io/git-branch: <branch1>`.
```
Comment (dockerymick) [inline applications/subscribe_git_resources.adoc]: In the following example,..

File: applications/subscribe_git_resources.adoc [hunk 3]
```diff
@@ -240,6 +241,72 @@ data:
...
+  name: git-mongodb-subscription
+  annotations:
+    apps.open-cluster-management.io/git-path: stable/ibm-mongodb-dev
+    apps.open-cluster-management.io/git-branch: <branch1>
+----
+
+[#subscribing-specific-commit]
+=== Subscribing to a specific commit
+
+The subscription operator that is included in this `multicloud-operators-subscription` repository subscribes to the latest commit of specified branch of a Git repository by default. If you want to subscribe to a specific commit, you need to specify the desired commit annotation with the commit hash in the subscription.
```
Comment (dockerymick) [inline applications/subscribe_git_resources.adoc]: Change "this" to "the"

File: applications/subscribe_git_resources.adoc [hunk 4]
```diff
@@ -240,6 +241,72 @@ data:
...
+
+The following example, the YAML file displays how to specify a different commit with `apps.open-cluster-management.io/git-desired-commit: <full commit number>`.
```
Comment (dockerymick) [inline applications/subscribe_git_resources.adoc]: In the following example,...

File: applications/subscribe_git_resources.adoc [hunk 5]
```diff
@@ -240,6 +241,72 @@ data:
...
+    apps.open-cluster-management.io/git-desired-commit: <full commit number>
+    apps.open-cluster-management.io/git-clone-depth: 100
+----
+
+The `git-clone-depth` annotation is optional and set to `20` by default, which means the subscription controller retrieves the previous 20 commit histories from the Git repository. If you specify a much older `git-desired-commit`, you need to specify `git-clone-depth` accordingly for the desired commit.
+
+[#subscribing-specific-tag]
+=== Subscribing to a specific tag
+
+The subscription operator that is included in this `multicloud-operators-subscription` repository subscribes to the latest commit of specified branch of a Git repository by default. If you want to subscribe to a specific tag, you need to specify the tag annotation in the subscription.
```
Comment (dockerymick) [inline applications/subscribe_git_resources.adoc]: Change "this" to "the"

File: applications/subscribe_git_resources.adoc [hunk 6]
```diff
@@ -240,6 +241,72 @@ data:
...
+
+The following example, the YAML file displays how to specify a different tag with `apps.open-cluster-management.io/git-tag: <v1.0>`:
```
Comment (dockerymick) [inline applications/subscribe_git_resources.adoc:292]: In the following example,...

**Review by dockerymick — APPROVED** [review]
Made a few suggestions, small edits

**Review by rokej — APPROVED** [review]
Thank you

File: applications/subscribe_git_resources.adoc (hunk 1 above)
Comment (swopebe) [reply applications/subscribe_git_resources.adoc:253]: Right, should not be this. Changed

File: applications/subscribe_git_resources.adoc (hunk 6 above)
Comment (swopebe) [reply applications/subscribe_git_resources.adoc:292]: This is correct, as without the comma, the subject changes. Not fully certain if that is what you are raising, though some did need colons, which I added.

**Comment by swopebe** [comment]
Thanks, all!

**Pull request merged**